src/components/ComponentPreview.stories.tsx

```diff
@@ -11,7 +12,9 @@ export default {
 type Story = StoryObj<typeof ComponentPreview>;
 
 const Template: StoryFn<typeof ComponentPreview> = ({component}) => (
-  <ComponentPreview component={component} />
+  <Formik initialValues={component} onSubmit={() => {}}>
+    <ComponentPreview component={component} />
+  </Formik>
```

Comment:
My first thought was that this should use the [`withFormik`](https://github.com/open-formulieren/formio-builder/blob/main/.storybook/decorators.tsx#L98`) decorator, but then you'd have to change all the stories to pass the same component via args and parameters for the decorator. That makes me believe we should not be enforcing formik in the preview wrapper component, but probably better the layer above. Instead, make the preview component take an extra prop `onJSONChange` or something like that. In the stories you can then easily provide the `fn()` arg for it.

I think this makes sense, because the JSON editor itself is agnostic about Formik, and so is `ComponentPreview` - I was also planning to re-use bits of this in another render context that may not have any form state. We might also expose a read-only JSON preview variant in case JSON editing gets tied to certain user permissions, and then the formik context also doesn't make a lot of sense.

Reply:
Made it so we don't rely on formik here.

> We might also expose a read-only JSON preview variant in case JSON editing gets tied to certain user permissions, and then the formik context also doesn't make a lot of sense.

Our `JSONEditor` supports a `readOnly` prop, so this could be easily doable in the future. The `onChange` prop might need to be made optional.

Reply:
Good to know the options for read only are there - let's make those changes when we decide to proceed with this and park that for now :)